controllers/configurationpolicy_controller.go

```diff
@@ -1110,22 +1125,27 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 				relatedObjects = updateRelatedObjectsStatus(relatedObjects, object)
 			}
 		}
-		// violations for enforce configurationpolicies are already handled in handleObjects,
-		// so we only need to generate a violation if the remediationAction is set to inform
-		if !handled && !enforce {
+
+		if mergeMessageEnforce || (!handled && !enforce) {
 			objData := map[string]interface{}{
 				"indx":        indx,
 				"kind":        kind,
 				"desiredName": templateObjs[indx].name,
 				"namespaced":  templateObjs[indx].isNamespaced,
 			}
 
-			statusUpdateNeeded := createInformStatus(
+			statusUpdateNeeded := createMergedStatus(
 				objShouldExist, numCompliant, numNonCompliant, compliantObjects, nonCompliantObjects, &plc, objData,
 			)
+
 			if statusUpdateNeeded {
 				parentStatusUpdateNeeded = true
 			}
+
+			if mergeMessageEnforce {
+				r.Recorder.Event(&plc, eventNormal, fmt.Sprintf(plcFmtStr, plc.GetName()),
```

Comment:
Is this correct sending event? is this message will attach to policy history? curious